### PR TITLE
fea: transform alert from enum lib into diagnostic

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | 
| ❓ Documentation  | no 

## Description

transform alert from enum lib into diagnostic when using the facade implementing the new api

### CODE REVIEW

File: enumeration/alerter/alert.go [hunk 1]
```diff
@@ -22,6 +27,10 @@ func (f *FakeAlert) ShouldIgnoreResource() bool {
 	return f.IgnoreResource
 }
 
+func (f *FakeAlert) Resource() *resource.Resource {
```
Comment (craigfurman) [inline enumeration/alerter/alert.go:30]: Is this FakeAlert still used anywhere?

File: enumeration/diagnostic/diagnostic.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+type diagnosticImpl struct {
+	alert alerter.Alert
+}
+
+func (d *diagnosticImpl) Code() string {
+	if _, ok := d.alert.(*alerts.RemoteAccessDeniedAlert); ok {
+		return "ACCESS_DENIED"
+	}
+	return "NOT_IMPLEMENTED"
```
Comment (craigfurman) [inline enumeration/diagnostic/diagnostic.go:24]: Is this user-facing? If so, wouldn't something like UNKNOWN_ERROR be more appropriate?

**Review by craigfurman — DISMISSED** [review]
2 non-blocking notes, I'll approve in advance 👍 

Before we merge this, I/you/we could test this branch out in snyk-iac-test and see if we can spike out bubbling up these non-fatal errors from diagnostics. WDYT?

File: enumeration/alerter/alert.go (hunk 1 above)
Comment (eliecharra) [reply enumeration/alerter/alert.go:30]: Yeah in some tests, it's a manual mock if you see what I mean.

**Comment by eliecharra** [comment]
> Before we merge this, I/you/we could test this branch out in snyk-iac-test and see if we can spike out bubbling up these non-fatal errors from diagnostics. WDYT?

100% in sync with that, let's do a small pairing session on that, shoot me an invite 🙏🏻

File: enumeration/diagnostic/diagnostic.go (hunk 2 above)
Comment (eliecharra) [reply enumeration/diagnostic/diagnostic.go:24]: Yeah that's user facing here, I didn't want to return an empty value but `UNKNOWN_ERROR` you proposed is probably better 👍🏻 Will change that thanks

File: enumeration/alerter/alert.go (hunk 1 above)
Comment (craigfurman) [reply enumeration/alerter/alert.go:30]: Shouldn't we move it into the test files then?

Comment (eliecharra) [reply enumeration/alerter/alert.go:30]: Yeah agreed, that's typically tech debt here.

Comment (craigfurman) [reply enumeration/alerter/alert.go:30]: you're right, it's out of scope for this PR! I forgot I was commenting on the new interface method, not a whole new struct!

**Review by craigfurman — DISMISSED** [review]
Nice!

**Pull request merged**